test/functional/ui/highlight_spec.lua

```diff
@@ -16,6 +17,91 @@ describe('color scheme compatibility', function()
   end)
 end)
 
+describe('buffer switching leaves manual syntax highlights ok', function()
+  -- When using manual syntax highlighting, it should be preserved even when
+  -- switching buffers... bug did only occur without :set hidden
+  -- Ref: vim patch 7.4.1236
+  local screen
+
+  before_each(function()
+    clear()
+    screen = Screen.new()
```

Comment:
Consider setting a smaller screen size here so the screen tests don't take up as much space.